train.py

```diff
@@ -133,19 +135,27 @@ def _fn(*args):
                 val_loss += val_loss_t.item()
                 val_cce += val_cce_t.item()
 
-                token_loss, grads = grad_norm_fn(variables, val_key, xt, xtp1)
-                grad_norm.append((xt, xtp1, token_loss, grads, step))
+                # TODO: deprecated for now
+                # token_loss, grads = grad_norm_fn(variables, val_key, xt, xtp1)
+                token_loss, _ = grad_norm_fn(variables, val_key, xt, xtp1)
+                # misc_metrics.append((xt, xtp1, token_loss, grads, step))
+                misc_metrics.append((xt, xtp1, token_loss, step))
 
             log_dict["val/loss"] = val_loss / n_val_steps
             log_dict["val/cce"] = val_cce / n_val_steps
 
             # CFG accuracy eval:
-            gen_tokens = gen_fn(variables, val_key)
-            tot_cfg_samples = sum((tokenizer.decode(t).split(',')[1:-1] for t in gen_tokens), start=[])
-            print(len(tot_cfg_samples))
-
-            cfg_acc = sum([cfg.verify(s) for s in tot_cfg_samples]) / len(tot_cfg_samples)
-            log_dict["val/cfg_acc"] = cfg_acc
+            # TODO: this needs to be removed into some "eval suite" which you configure
+            # if cfg is not None: 
+            #     gen_tokens = gen_fn(variables, val_key)
+            #     tot_cfg_samples = sum((tokenizer.decode(t).split(',')[1:-1] for t in gen_tokens), start=[])
+
+            #     cfg_acc = sum([cfg.verify(s) for s in tot_cfg_samples]) / len(tot_cfg_samples)
+            #     log_dict["val/cfg_acc"] = cfg_acc
+            # else: # just sample some and print
+            #     gen_tokens = gen_fn(variables, val_key)
+            #     gen_samples = [tokenizer.decode(t) for t in gen_tokens]
+            #     print(gen_samples[:5])
```

Comment:
I'd recommend removing the above altogether and bring it back in the form of some "eval" fn which is configured externally and is just a 1 line call to some other file